Updated tracks tutorial to support Agent (#143414)
elastic/kibana: docs/maps/asset-tracking-tutorial.asciidoc

You’ll learn to:

> - Use the Elastic Agent to ingest the TriMet REST API into Elasticsearch.

Reviewer: ```suggestion
- Use {agent} to ingest the TriMet REST API into {es}.
```

- If you don’t already have {kib}, set it up with https://www.elastic.co/cloud/elasticsearch-service/signup?baymax=docs-body&elektra=docs[our free trial]. Download the deployment credentials.
- Obtain an API key for https://developer.trimet.org/[TriMet web services] at https://developer.trimet.org/appid/registration/.

> - {fleet-guide}/fleet-overview.html[Fleet] is enabled on your cluster, and one or more {fleet-guide}/elastic-agent-installation.html[Elastic Agents] is enrolled.

Reviewer: ```suggestion
- {fleet-guide}/fleet-overview.html[Fleet] is enabled on your cluster, and one or more {fleet-guide}/elastic-agent-installation.html[{agent}s] is enrolled.
```

[float]
=== Part 1: Ingest the Portland bus data

> To get to the fun of visualizing and alerting on Portland buses, you must first add the *Custom API* integration to an Elastic Agent policy to get the TriMet Portland bus data into {es}.

Author: ```suggestion
To get to the fun of visualizing and alerting on Portland buses, you must first add the *Custom API* integration to an Elastic Agent policy, if one doesn't already exist click *Create agent policy* and give it a name to create one, this will then be used to get the TriMet Portland bus data into {es}.
```

> To get to the fun of visualizing and alerting on Portland buses, you must first add the 'Custom API' integration to an Elastic Agent policy to get the TriMet Portland bus data into {es}.

Reviewer: ```suggestion
To get to the fun of visualizing and alerting on Portland buses, you must first add the *Custom API* integration to an Elastic Agent policy to get the TriMet Portland bus data into {es}.
```

----------------------------------

[float]

> ==== Step 2: Configure Elastic Agent

Reviewer: ```suggestion
==== Step 2: Configure {agent}
```

. From the {kib} main menu, click *Fleet*, then the *Agent policies* tab.

> . Click the name of the agent policy where you want to add the *Custom API* integration. The configuration changes you make only apply to the policy you select.

Reviewer 2: should we expect users to have an agent policy? What if there are no agent policies?

Reviewer 3: Then they will have to create one.

Author: If there are no policies it will prompt them to create one. Will update to reflect.

> . Click the name of the agent policy where you want to add the *Custom API* integration to. The configuration changes you make only apply to the policy you select.

Reviewer: ```suggestion
. Click the name of the agent policy where you want to add the *Custom API* integration. The configuration changes you make only apply to the policy you select.
```

> . Click the name of the *Custom API* integration, or add the integration first if the agent policy does not yet have it.

Reviewer: ```suggestion
. Click the name of the *Custom API* integration, or add the integration if the agent policy does not yet have it.
```

. From the *Edit Custom API integration* page, expand the *Change defaults* section.

> . Set the *Dataset name* to *httpjson.trimet*.

Reviewer 4: Maybe I don't understand Fleet very well, but isn't this the name of the index (e.g. `tri_met_tracks`)?

. Set the *Ingest Pipeline* to *tri_met_pipeline*.

. Set the *Request URL* to *https://developer.trimet.org/ws/v2/vehicles?appID=<tri_met_app_id>*.

. Set *Response Split* to *target: body.resultSet.vehicle*.

> . At the bottom of the configuration expand *Advanced options*.

Reviewer: ```suggestion
. At the bottom of the configuration, expand *Advanced options*.
```

. Set *Processors* to:
+
[source,yaml]
----------------------------------
- decode_json_fields:
    fields: ["message"]
    target: "trimet"
----------------------------------

. Leave everything else as defaults.

> . Click *Save integration* to deploy the configuration to any {agent} with the policy assigned.

Reviewer 2: How will this work if multiple agents try to ingest from REST API?

Reviewer 3: We are going to have data duplication.

> . Click *Save integration*, this will deploy the configuration to any Agents with the policy assigned.

Reviewer: ```suggestion
. Click *Save integration* to deploy the configuration to any {agent} with the policy assigned.
```
